src/DataAPI.jl

```diff
@@ -287,4 +287,157 @@ using a `sink` function to materialize the table.
 """
 function allcombinations end
 
+const STYLE_INFO = """
+One of the uses of the metadata `style` is decision
+how the metadata should be propagated when `x` is transformed. This interface
+defines the `:none` style that indicates that metadata should not be propagated
+under transformations. All types supporting metadata allow at least this style.
```

Comment:
@nalimilan - maybe we should make this description more precise? Currently in DataFrames.jl I needed to make a decision when `:none` metadata is kept and it is kept only in two cases:
* `DataFrame` constructor;
* `copy`;

all other operations drop all `:none` metadata. So, essentially both table level and column level `:none` metadata are attached to a concrete instance of a table or its copies (this is a safest approach, i.e. making sure that indeed when metadata could be invalidated it is dropped). Are we in agreement here?

Reply:
Given no response I will make the definition more precise.